hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableCommand.scala

```diff
@@ -146,11 +146,12 @@ object CreateHoodieTableCommand {
     val partitionColumnNames = hoodieCatalogTable.partitionSchema.map(_.name)
     // Remove some properties should not be used;append pk, preCombineKey, type to the properties of table
     val newTblProperties =
-      hoodieCatalogTable.catalogProperties.--(needFilterProps) ++ HoodieOptionConfig.extractSqlOptions(properties)
+      hoodieCatalogTable.catalogProperties.--(needFilterProps) + ("provider" -> "hudi") ++  HoodieOptionConfig.extractSqlOptions(properties)
```

Comment:
Thanks for your contribution! Let's create a JIRA ticket to track this fix. Additionally, it would be valuable to implement a test that verifies all tables created by Hudi have the correct configuration settings applied.